src/main.cpp

```diff
@@ -0,0 +1,234 @@
+
+  NormalizedPoint() : n_x(0), n_y(0) {}
+};
+
+struct NormalizedConnection {
+  unsigned int id;
+  unsigned int start_element_id;
+  unsigned int end_element_id;
+  std::vector<NormalizedPoint> vertices;
+
+  NormalizedConnection() : id(-1), start_element_id(-1), end_element_id(-1) {}
+};
+
+struct NormalizedElement {
+  unsigned int id;
+  NormalizedPoint point;
+  float n_w;
+  float n_h;
+  std::vector<NormalizedConnection> connections;
+
+  NormalizedElement() : id(-1), n_w(0), n_h(0) {}
+};
+
+float normalizedToScreenX(const float n_x, const int screen_w) {
+  return n_x * screen_w;
+}
+
+float normalizedToScreenY(const float n_y, const int screen_h) {
+  return n_y * screen_h;
+}
+
+void drawBackground(SDL_Renderer *renderer) {
+  SDL_SetRenderDrawColor(renderer, 0, 0, 0, SDL_ALPHA_OPAQUE);
+  SDL_RenderClear(renderer);
+}
+
+int parseInput(
+    const char *filename,
+    std::vector<NormalizedElement> &elements_to_parse) {
+  pugi::xml_document file;
+  if (!file.load_file(filename)) {
+    return PARSER_FAILURE;
+  }
+
+  pugi::xml_node elements = file.child("logic_scheme").child("elements");
+
+  //Parsing elements from given file
+  for (pugi::xml_node element = elements.first_child();
+      element;
+      element = element.next_sibling()) {
+    NormalizedElement parsed_element;
+    parsed_element.id = atoi(element.attribute(parser_e_id).value());
+    parsed_element.point.n_x = std::stof(element.attribute(parser_x).value());
+    parsed_element.point.n_y = std::stof(element.attribute(parser_y).value());
+    parsed_element.n_h = std::stof(element.attribute(parser_height).value());
+    parsed_element.n_w = std::stof(element.attribute(parser_width).value());
+
+    //Parsing connections for given element
+    for (pugi::xml_node connection = element.first_child();
+        connection;
+        connection = connection.next_sibling()) {
+      NormalizedConnection parsed_connection;
+      parsed_connection.id = atoi(connection.attribute(parser_c_id).value());
+      parsed_connection.start_element_id = parsed_element.id;
+      parsed_connection.end_element_id = atoi(connection.attribute(parser_end_elem).value());
+
+      //Parsing vertices for given connection
+      for (pugi::xml_node vertex = connection.first_child();
+          vertex;
+          vertex = vertex.next_sibling()) {
+        NormalizedPoint parsed_vertex;
+        parsed_vertex.n_x = std::stof(vertex.attribute(parser_x).value());
+        parsed_vertex.n_y = std::stof(vertex.attribute(parser_y).value());
+        parsed_connection.vertices.push_back(parsed_vertex);
+      }
+      parsed_element.connections.push_back(parsed_connection);
+    }
+    elements_to_parse.push_back(parsed_element);
+  }
+  return 0;
+}
+
+std::ostream &operator<<(
+    std::ostream &out,
+    const NormalizedElement &element_to_print) {
+  out << "Element id: " << element_to_print.id << " x: " <<
+  element_to_print.point.n_x << " y: " <<
+  element_to_print.point.n_y << std::endl;
```

Comment:
Consider formatting multi-line statements:
```suggestion
  out << "Element id: " << element_to_print.id
      << " x: " <<  element_to_print.point.n_x
      << " y: " <<  element_to_print.point.n_y
      << std::endl;
```